services/src/api/model/db_types.rs

```diff
@@ -0,0 +1,408 @@
+                value: Some(value),
+                attribute: None,
+                factor: None,
+                default_value: None,
+            } => Ok(Self::Static {
+                value: value as usize,
+            }),
+            NumberParamDbType {
+                value: None,
+                attribute: Some(attribute),
+                factor: Some(factor),
+                default_value: Some(default_value),
+            } => Ok(Self::Derived(DerivedNumber {
+                attribute,
+                factor,
+                default_value,
+            })),
+            _ => Err(Error::UnexpectedInvalidDbTypeConversion),
+        }
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "Symbology")]
+pub struct SymbologyDbType {
+    raster: Option<RasterSymbology>,
+    point: Option<PointSymbology>,
+    line: Option<LineSymbology>,
+    polygon: Option<PolygonSymbology>,
+}
+
+impl From<Symbology> for SymbologyDbType {
+    fn from(symbology: Symbology) -> Self {
+        match symbology {
+            Symbology::Raster(raster) => SymbologyDbType {
+                raster: Some(raster),
+                point: None,
+                line: None,
+                polygon: None,
+            },
+            Symbology::Point(point) => SymbologyDbType {
+                raster: None,
+                point: Some(point),
+                line: None,
+                polygon: None,
+            },
+            Symbology::Line(line) => SymbologyDbType {
+                raster: None,
+                point: None,
+                line: Some(line),
+                polygon: None,
+            },
+            Symbology::Polygon(polygon) => SymbologyDbType {
+                raster: None,
+                point: None,
+                line: None,
+                polygon: Some(polygon),
+            },
+        }
+    }
+}
+
+impl TryFrom<SymbologyDbType> for Symbology {
+    type Error = Error;
+
+    fn try_from(symbology: SymbologyDbType) -> Result<Self, Self::Error> {
+        match symbology {
+            SymbologyDbType {
+                raster: Some(raster),
+                point: None,
+                line: None,
+                polygon: None,
+            } => Ok(Self::Raster(raster)),
+            SymbologyDbType {
+                raster: None,
+                point: Some(point),
+                line: None,
+                polygon: None,
+            } => Ok(Self::Point(point)),
+            SymbologyDbType {
+                raster: None,
+                point: None,
+                line: Some(line),
+                polygon: None,
+            } => Ok(Self::Line(line)),
+            SymbologyDbType {
+                raster: None,
+                point: None,
+                line: None,
+                polygon: Some(polygon),
+            } => Ok(Self::Polygon(polygon)),
+            _ => Err(Error::UnexpectedInvalidDbTypeConversion),
+        }
+    }
+}
+
+macro_rules! delegate_from_to_sql {
```

Comment:
maybe write a quick comment for the future generation here what this does?
"Implements To/FromSql for a type by means of an intermediary database type. This is for cases where a Rust struct or enum cannot directly be represented by Postgres." Oder so.